### ISSUE TITLE
Latest version of cryptography (41) leads to solver inconsistences

### ISSUE THREAD

**Issue by hmaarrfk** [conda-forge/pyopenssl-feedstock#30]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

It seems that the latest version of cryptography is out (41) which leads the solver to not know which one to pick.

```
mamba install "pyopenssl>=23"

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (1.4.2) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Looking for: ["pyopenssl[version='>=23']"]

conda-forge/linux-64                                        Using cache
conda-forge/noarch                                          Using cache

Pinned packages:
  - python 3.9.*


Transaction

  Prefix: /home/mark/mambaforge/envs/m

  Updating specs:

   - pyopenssl[version='>=23']
   - ca-certificates
   - certifi
   - openssl


  Package         Version  Build           Channel                    Size
────────────────────────────────────────────────────────────────────────────
  Upgrade:
────────────────────────────────────────────────────────────────────────────

  - pyopenssl      21.0.0  pyhd8ed1ab_0    conda-forge                    
  + pyopenssl      23.1.1  pyhd8ed1ab_0    conda-forge/noarch       Cached

  Downgrade:
────────────────────────────────────────────────────────────────────────────

  - cryptography   41.0.0  py39hd4f0224_0  conda-forge                    
  + cryptography   40.0.2  py39h079d5ae_0  conda-forge/linux-64     Cached

```

### Installed packages

```shell
the above
```

### Environment info

```shell
linux
```

**Comment by h-vetinari** [comment]
Very simply the cryptography cap in this recipe should not be present. Upstream [doesn't](https://github.com/pyca/pyopenssl/blob/main/setup.py#L99-L103) have a cap either

**Comment by h-vetinari** [comment]
> Upstream [doesn't](https://github.com/pyca/pyopenssl/blob/main/setup.py#L99-L103) have a cap either

My apologies, I just saw the `!40` portions. There's actually a cap upstream as well 🤦

**Comment by h-vetinari** [comment]
New version has been [tagged](https://github.com/pyca/pyopenssl/releases/tag/23.2.0), you can either wait until the builds reach PyPI and the bot opens a PR, or switch to GH sources if you're in a hurry. :)

**Comment by h-vetinari** [comment]
Not just solver inconsistencies, it can even make conda environments unusable, because all conda calls want to check the channel metadata, and if (py)openssl is broken, then you're done

**Comment by h-vetinari** [comment]
This has hopefully been solved by #31.

**Comment by xhochy** [comment]
Also needs #32
